commit 910c9e41186762de3717baaf392ab5ff0c454496
Author: Eugene Teo <eteo@redhat.com>
Date:   Mon Apr 13 10:04:41 2009 +0800

    unreached code in selinux_ip_postroute_iptables_compat() (CVE-2009-1184)
    
    Not upstream in 2.6.30, as the function was removed there, making this a
    non-issue.
    
    Node and port send checks can skip in the compat_net=1 case. This bug
    was introduced in commit effad8d.
    
    Signed-off-by: Eugene Teo <eugeneteo@kernel.sg>
    Reported-by: Dan Carpenter <error27@gmail.com>
    Acked-by: James Morris <jmorris@namei.org>
    Acked-by: Paul Moore <paul.moore@hp.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

Adjusted to apply to Debian's 2.6.26 by dann frazier <dannf@debian.org>

diff -urpN linux-source-2.6.26.orig/security/selinux/hooks.c linux-source-2.6.26/security/selinux/hooks.c
--- linux-source-2.6.26.orig/security/selinux/hooks.c	2008-07-13 15:51:29.000000000 -0600
+++ linux-source-2.6.26/security/selinux/hooks.c	2009-05-03 12:29:51.000000000 -0600
@@ -4416,6 +4416,7 @@ static int selinux_ip_postroute_iptables
 	if (err)
 		return err;
 	err = avc_has_perm(sk_sid, if_sid, SECCLASS_NETIF, netif_perm, ad);
+	if (err)
 		return err;
 
 	err = sel_netnode_sid(addrp, family, &node_sid);
